### ISSUE TITLE
server.go:11:2: no required module provides package github.com/gorilla/mux: working directory is not part of a module[bug] 

### ISSUE THREAD

**Issue by nachi2609** [gorilla/mux#622]
Installed mux exactly the way it is mentioned but getting this error.

**Comment by lmakarov** [comment]
A go app first needs to be initialized with `go mod init` before you can use external modules. This should help:

```
go mod init myapp
go get -u github.com/gorilla/mux
go mod tidy
```

**Comment by samvolton** [comment]
thanks bro

**Comment by nachi2609** [comment]
got it

**Comment by pheetah** [comment]
> A go app first needs to be initialized with `go mod init` before you can use external modules. This should help:
> 
> ```
> go mod init myapp
> go get -u github.com/gorilla/mux
> go mod tidy
> ```

You are awesome. Thank you so much!

**Comment by aldipoin26** [comment]
thank you so much. i actually search for his for a few hours before finding this

**Comment by aisha-razzaq** [comment]
by add this code, vscode is giving me error: "main redeclared in this block", any solution?

**Comment by Rafaelobn** [comment]
It isn't working. I already did it, the exacly way

> A go app first needs to be initialized with `go mod init` before you can use external modules. This should help:
> 
> ```
> go mod init myapp
> go get -u github.com/gorilla/mux
> go mod tidy
> ```

I have a go.mod with tue github.com/gorilla/mux on it... I tried deleting it, and did the "go get -u github.com/gorilla/mux" again. It recreats the mux line in the go.mod file, but it presents the exacly same problem. 
![Problema](https://github.com/user-attachments/assets/690c09ce-0606-402b-8852-b26ccf1b9ab2)

Doing go run main.go returns the exacly same problema.
